src/libraries/System.Text.RegularExpressions/gen/DiagnosticDescriptors.cs

```diff
@@ -62,5 +62,13 @@ internal static class DiagnosticDescriptors
             category: Category,
             DiagnosticSeverity.Info,
             isEnabledByDefault: true);
+
+        public static DiagnosticDescriptor UseRegexSourceGeneration { get; } = new DiagnosticDescriptor(
+            id: RegexAnalyzer.DiagnosticId,
```

Comment:
It could, but I do reference it in the fixer since it needs to specify which diagnostics does it fix, which is why I wanted to avoid having the string duplicated in the code and instead just centralized. That said, I can have it duplicated if that is prefered.

Reply:
Where you need the ID, you can't just do DiagnosticDescriptors.UseRegexSourceGeneration.Id?

Reply:
oh, of course, not sure why I didn't think of that, will fix.